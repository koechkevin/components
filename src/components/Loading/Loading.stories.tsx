import { storiesOf } from '@storybook/react';
import React from 'react';

import Loading from './Loading';

storiesOf('Loading', module).add('default', () => <Loading spinning />);
