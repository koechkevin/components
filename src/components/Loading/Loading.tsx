import { Icon, Spin } from 'antd';
import NProgress from 'nprogress';
import React, { FC, useEffect } from 'react';

import styles from './Loading.module.scss';

NProgress.configure({ showSpinner: false });

interface Props {
  spinning?: boolean;
}

export const Loading: FC<Props> = (props) => {
  const { spinning } = props;
  const antIcon = <Icon type="loading-3-quarters" spin />;

  useEffect(() => {
    NProgress.start();
    return () => {
      NProgress.done();
    };
  }, []);

  return <div className={styles.loading}>{spinning && <Spin size="large" indicator={antIcon} />}</div>;
};

export default Loading;
