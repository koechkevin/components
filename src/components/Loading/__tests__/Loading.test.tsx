import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import Loading from '../Loading';

describe('Loading Component Test', () => {
  it('renders with loading icon', () => {
    const tree = shallow(<Loading />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders with tip', () => {
    const tree = shallow(<Loading tip="Please wait, loading..." />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
