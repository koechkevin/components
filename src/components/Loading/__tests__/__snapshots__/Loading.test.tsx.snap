// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Test renders with loading icon 1`] = `
<div
  className="loading"
/>
`;

exports[`Loading Component Test renders with tip 1`] = `
<div
  className="loading"
/>
`;
